packages/unmock-core/src/generator.ts

```diff
@@ -3,23 +3,46 @@
  */
 import {
   CreateResponse,
-  GeneratedMock,
-  IMock,
   ISerializedRequest,
-  RequestToSpec,
+  ISerializedResponse,
+  IServiceDef,
+  IServiceDefLoader,
 } from "./interfaces";
+
+import { IService } from "./service/interfaces";
+
 import { Schema, UnmockServiceState } from "./service/interfaces";
 
 // json-schema-faker doesn't have typed definitions?
 // @ts-ignore
 import jsf from "json-schema-faker";
+import { ServiceParser } from "./service/parser";
+import { ServiceStore } from "./service/serviceStore";
 
-export const responseCreatorFactory = (): CreateResponse => {
-  return (__: ISerializedRequest) => ({
-    body: "Nothing here yet",
-    statusCode: 200,
-  });
-};
+export function responseCreatorFactory({
+  serviceDefLoader,
+}: {
+  serviceDefLoader: IServiceDefLoader;
+}): CreateResponse {
+  const serviceDefs: IServiceDef[] = serviceDefLoader.loadSync();
+  const parser = new ServiceParser();
+  const services: IService[] = serviceDefs.map(serviceDef =>
+    parser.parse(serviceDef),
+  );
+  const serviceStore = new ServiceStore(services);
+  return (sreq: ISerializedRequest) => {
+    const responseTemplateOrUndefined = serviceStore.match(sreq);
+
+    if (responseTemplateOrUndefined === undefined) {
+      return undefined;
+    }
+
+    const responseTemplate = responseTemplateOrUndefined;
+
+    const sres = generateMockFromTemplate(responseTemplate);
+    return sres;
```

Comment:
I'd assume either:
1. `generateMockFromTemplate` returns a valid response even for an empty template or undefined (i.e. empty body?)
2. Preferably, if the `ServiceStore` can't match, I'd expect an error rather than failing silently (following the "fail to success" paradigm).

Reply:
Response creator is currently defined as returning undefined if it cannot return a response, and it's the backend's problem to deal with that (as it currently does by capturing the error and returning 501).

Reply:
I'm not a big fan of either of those; ingesting a '501' and understanding it's an unmock-related error is not trivial at all.
I think it's a `core` issue, and the backends can catch the error if they so wish, but `ServiceStore` not being able to match a request is quite severe IMO

Reply:
Of course that's not how we want to handle the error, but that's again another PR to fix how to handle that.

Reply:
In Java or Scala world `CreateResponse` would, imho, return `Option<Response>`. In TypeScript we use `undefined` for the empty option. In my opinion it does not make sense to throw for an event that's very much expected to happen often (no match), but instead just return empty in that case and let the outer edge of the application handle that.

Reply:
Well then the core issue is how often do we expect this to happen? I think it's not an expected outcome nor one that should happen often.
If you're using GitHub, you should have the GitHub service specification. If you're making a call to an endpoint that doesn't exist on GitHub - it's probably a typo.
If you're using an SDK and you have the service specification - I don't expect there to be any such mistakes (assuming the service specification is up to date).

The only time it would happen frequently IMO is when someone will want to build their service from the ground up?

Reply:
Yes, and we're still throwing an error for that event at the outer edge of our application. It's just that internally it's not an application error if no match is found so I don't see any reason to throw, it's user error and expected to happen.

Reply:
Well then maybe the best compromise would be to at least add type guards so that any backends that use `responseCreatorFactory` have to return a `ISerializedResponse`, so that any `undefined` is dealt with and not simply returned?

Reply:
One cannot return `ISerializedResponse | undefined` from any backend, for Node.js one must construct valid `http.ServerResponse` from it and that's not possible to do without first checking if it's undefined. The compiler will barf with `Argument of type 'ISerializedResponse | undefined' is not assignable to parameter of type 'ISerializedResponse'.` if that check is removed.